### PR TITLE
feat: add search api of graphql

### PR DESCRIPTION
**Pull request by Naupio**
## What problem does this PR solve?
warning: this pr depend on https://github.com/Magickbase/godwoken_explorer/pull/1227

issue: #1231 
add search api of graphql
example
    keyword can be: udt name| account eth_address | address | transaction hash | block number
```graphql
query {
  search_keyword(input: { keyword: "UDT"}){
    type
    id
  }
}
```
#### Test
- unit test
#### Task
 - none

### CODE REVIEW

**Comment by FrederLu** [comment]
image
image
image
When using Bridged Token and ERC20/ERC721/ERC1155 names, it will prompt that no information can be found.

**Comment by Naupio** [comment]
@Keith-CY  @JeffreyMa597 we need discuss some feat of this PR.

previously, we have `search_udt` api for search udt with fuzzy_name that returning a pagination list resut `paginate_search_udts object` 

in here, we add graphql api `search_keyword` to replace `restful api` of search. but the returning object is `search_result` witch with `type` and `id` field for redirect the next page.

there a problem of using `search_keyword` api with `UDT` name which have more than one record will returning not_found like this
```
  "errors": [
    {
      "locations": [
        {
          "column": 3,
          "line": 17
        }
      ],
      "message": "not_found",
      "path": [
        "search_keyword"
      ]
    }
```

**Comment by Naupio** [comment]
query example of testnet-stg env
```graphql
query {
  search_udt(input: { fuzzy_name: "%ERC%", limit: 1 }) {
    entries {
      id
      name
      symbol
      type
      contract_address_hash
    }
    metadata {
      total_count
      before
      after
    }
  }

  search_keyword(
    input: { keyword: "%MSHKUUPS%" }
  ) {
    type
    id
  }
}
```
```json
{
  "data": {
    "search_keyword": null,
    "search_udt": {
      "entries": [
        {
          "contract_address_hash": "0xfa5a662b13208070e665ac8239bff9ad6e78a020",
          "id": 92081,
          "name": "MSHK ERC721 UUPS Token",
          "symbol": "MSHKUUPS",
          "type": "NATIVE"
        }
      ],
      "metadata": {
        "after": "g3QAAAABZAACaWRiAAFnsQ==",
        "before": null,
        "total_count": 808
      }
    }
  },
  "errors": [
    {
      "locations": [
        {
          "column": 3,
          "line": 17
        }
      ],
      "message": "not_found",
      "path": [
        "search_keyword"
      ]
    }
  ]
}
```

**Review by FrederLu — APPROVED** [review]

**Comment by Keith-CY** [comment]
@zmcNotafraid please have a review

**Comment by Keith-CY** [comment]
As mentioned in https://github.com/Magickbase/godwoken_explorer/issues/618#issuecomment-1365007213, if the keyword is the address of an uninitialized account, `Address` but not `Account` will be returned in json API. What would it be like in the graphql API if the address is of an uninitialized account.

**Comment by FrederLu** [comment]
Is an uninitialized account an address that doesn't exist? If you use a non-existent address graphql prompts not found.
image

**Comment by Keith-CY** [comment]
> Is an uninitialized account an address that doesn't exist? If you use a non-existent address graphql prompts not found. image

TL;DR;
An uninitialized account is an account/address that doesn't exist on-chain. It would be `null/not found` if an uninitialized account appears in response but now info of an `uninitialized account/address` is expected to be returned within the response(most fields are empty except the .bit domain).

---
Before [this feature](https://github.com/Magickbase/godwoken_explorer/issues/618#issuecomment-1365007213), if an account hasn't sent a transaction, it will not be initialized on the chain and it will not be stored in the database of gwscan. So the API returns `not found`. This case leads to a potential issue that `.bit` domain cannot be returned along with some APIs. Let's say that we have a request for a transaction and the `to` is an initialized account. The response would be 

```
from_account: From Account
from_address: 0x.....
to_account: null
to_address: 0x......
```

If `to_address` has a corresponding `.bit domain`, it cannot be returned within `to_account` because it's `null`

[This feature](https://github.com/Magickbase/godwoken_explorer/issues/618#issuecomment-1365007213) changed the behavior of gwscan. Even if the account has not sent a transaction(only appears in `to` of a transaction), it will still be stored in the database of gwscan and the response of requesting the transaction above would be

```
from_account: From Account
from_address: 0x.....
to_account: To Account
to_address: 0x.....
```

Now `to`'s `.bit domain` could be returned within `To Account`.

**Review by zmcNotafraid — APPROVED** [review]

**Comment by Naupio** [comment]
warning: change `account` api result to union type 'account_or_address'

example
```
    query {
      account(input: {address: "0x59b670e9fa9d0a427751af201d676719a970857b"}){
        ... on Account{
          type
          eth_address
        }
      }
    }
```
